src/test/java/brokeculator/parser/AddParserTest.java

```diff
@@ -0,0 +1,41 @@
+package brokeculator.parser;
+
+import brokeculator.command.AddCommand;
+import brokeculator.command.Command;
+import brokeculator.command.InvalidCommand;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+
+public class AddParserTest {
+    @Test
+    void parseInput_stringCorrectFormat_addCommand() {
+        String input = "add /n test1 /d 1900 /a 10 /c category1";
+        Command resultCommand = AddParser.parseInput(input);
+        assertInstanceOf(AddCommand.class, resultCommand);
+    }
+    @Test
+    void parseInput_stringNoCategory_addCommand() {
+        String input = "add /n test1 /d 1900 /a 10";
+        Command resultCommand = AddParser.parseInput(input);
+        assertInstanceOf(AddCommand.class, resultCommand);
+    }
+    @Test
+    void parseInput_stringInvalidFormat_invalidCommand() {
```

Comment:
Wouldn't we expect a AddCommand with "null" category? Also, is this not a valid format where the user inputs /c but without a category?